### DISCUSSION TITLE
Changing the public URLs of Images to another

### DISCUSSION THREAD

**Discussion by dulranga** [directus/directus#7700] [Archive]
I am using **Directus** for my blog. When I am working with `WYSIWYG` editor to make the content of my blog, I faced this issue.
The issue is it displays the back-end domain URL with images. 
like this 
> _http://localhost:8080/assets/450dd094-358c-4575-b421-982a98b33ddf_ 

#### This is really bad because I don't like to expose the CMS URL to public. (Security Risk)
 
just need to change the **Public** URL of Images. like this.

> http://images.mydomain.com/assets/450dd094-358c-4575-b421-982a98b33ddf 

What I can do for this?

**Comment by rijkvanzanten** [comment]
There's no configuration for that (yet). You can [create a custom hook](https://docs.directus.io/guides/api-hooks/) to dynamically find/replace those urls on read/write 👍🏻

**Reply by dulranga** [reply]
#### Can you give me a example hook for this?.
>The content will looks like this.
`<h1><img src=\"http://localhost:8080/assets/450dd094-358c-4575-b421-982a98b33ddf\" alt=\"Day Night\" width=\"96\" height=\"96\" /></h1>\n<h1>this is  hello world</strong></p>`

It would be great and helpful if you can. Btw I am new to Directus

**Reply by rijkvanzanten** [reply]
There are example hooks in the docs link I sent before 👍🏻  The main implementation of the find-and-replace is plain JS, nothing Directus specific about it

**Comment by salk52** [comment]
@dulranga did you manage to solve this problem? I have the same request...

**Comment by luciandavidescu** [comment]
Assuming that you use the webserver to point to the directus assets folder via the other location and now you also want this to reflect on the website. 
(unless there may a way to do this in the editor or on the frontend), in a flow, on each update you'll have to get the article content, find and replace the strings and then write it back to the database.

**Comment by salk52** [comment]
It would be really nice if there is ENV setting, USE_HOST_IN_URL or something like that, and then in that case editor would add https://www.example.com or not. I need links to be without protocol, host and port and it seems to me that it's not something complicated to implement...

**Reply by luciandavidescu** [reply]
yep, there's a PR on this #16111 . Another way is to store your variables in settings or in a singleton collection and pull them from there with a read data block
